live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/util/EventUtils.java

```diff
@@ -78,4 +85,27 @@ public static EventType getEventType(RepoEvent<DataAttributes<NodeResource>> eve
         }
         throw new LiveIngesterRuntimeException("Unsupported Repo event type " + event.getType());
     }
+
+    public static boolean isNotPredictionNodeEvent(RepoEvent<DataAttributes<NodeResource>> event)
+    {
+        return !PREDICTION_NODE_TYPE.equals(event.getData().getResource().getNodeType());
+    }
+
+    public static boolean isNotPredictionApplyEvent(RepoEvent<DataAttributes<NodeResource>> event)
+    {
+        Set<String> aspects = Optional.ofNullable(event.getData().getResource().getAspectNames()).orElse(Collections.emptySet());
+        if (aspects.contains(PREDICTION_APPLIED_ASPECT))
+        {
+            String actualPredictionTime = (String) event.getData().getResource().getProperties().get(PREDICTION_TIME_PROPERTY);
+            String beforePredictionTime = (String) Optional.ofNullable(event.getData().getResourceBefore())
+                    .map(NodeResource::getProperties)
+                    .map(properties -> properties.get(PREDICTION_TIME_PROPERTY))
+                    .orElse(null);
+
+            return actualPredictionTime != null && actualPredictionTime.equals(beforePredictionTime)
```

Comment:
I hadn't really thought about the situation where we remove the prediction time from the node. This is fine for now, but I think we'll revisit this when we work on https://hyland.atlassian.net/browse/ACS-7708

Reply:
I've updated the logic